crates/wasi/src/preview2/tcp.rs

```diff
@@ -103,23 +107,76 @@ impl HostInputStream for HostTcpSocketInner {
 
 const SOCKET_READY_SIZE: usize = 1024 * 1024 * 1024;
 
+pub(crate) struct TcpWriteStream {
+    stream: HostTcpSocketInner,
+    write_handle: Option<tokio::task::JoinHandle<anyhow::Result<()>>>,
```

Comment:
I think this'll want to be the AbortOnDrop handle (or something like that name) Pat recently added too

Reply:
Great point!